CodeForgeLabs/Peer2PeerLearning
# Enhance Comments and Votes APIs with Additional Features and Minor Polishing

This pull request introduces comprehensive improvements and enhancements to the CommentController and VoteController while refining related functionality for Questions, Answers, and Tags. The changes aim to provide a robust and consistent experience across the application's REST API.

Key Updates
CommentController Enhancements:

Added endpoint for retrieving comments associated with specific questions, both unordered and sorted by creation date.
Introduced user-centric comment retrieval with GET /api/comments/user/{userId}.
Refined comment creation, update, and deletion endpoints to handle edge cases more effectively.
Improved response consistency, including proper error handling with meaningful HTTP status codes.
VoteController Enhancements:

Unified endpoint for voting on questions and answers, simplifying the API while maintaining flexibility.
Enhanced retrieval of votes, including per-user vote insights for specific entities.
Introduced precise error handling for invalid entity types or non-existent entities.
QuestionController, AnswerController, and Tag Polishing:

Minor optimizations were implemented to streamline query performance and improve maintainability.
Enhanced validation and error-handling mechanisms for a more robust user experience.
Adjusted responses for consistency with the overall API standards.
Impact of Changes
Developers: Simplified API structure reduces redundancy, making integration and maintenance easier.
Users: Improved functionality enhances reliability and ensures a consistent experience.
Codebase: Refinements ensure better scalability and adherence to clean code principles.
Additional Information
This update is backward-compatible, ensuring no disruption for existing clients.

If you have any questions, suggestions, or additional features to propose, feel free to share!

How to Test:
CommentController:

Test all endpoints (POST, PUT, DELETE, and GET) using Postman or a similar tool.
Verify responses for valid and invalid inputs, including edge cases like non-existent IDs.
VoteController:

Test voting functionality on both questions and answers, ensuring votes are applied and retrieved correctly.
Validate error handling for invalid type values or non-existent IDs.
General:

Test related changes in Questions, Answers, and Tags for expected behavior without regressions.
